rasa/core/channels/twilio_voice.py

```diff
@@ -0,0 +1,342 @@
+        "Polly.Ewa",
+        "Polly.Maja",
+        "Polly.Ricardo",
+        "Polly.Vitoria",
+        "Polly.Camila-Neural",
+        "Polly.Cristiano",
+        "Polly.Ines",
+        "Polly.Carmen",
+        "Polly.Maxim",
+        "Polly.Tatyana",
+        "Polly.Conchita",
+        "Polly.Enrique",
+        "Polly.Miguel",
+        "Polly.Penelope",
+        "Polly.Lupe-Neural",
+        "Polly.Astrid",
+        "Polly.Filiz",
+        "Polly.Gwyneth",
+    ]
+
+    SUPPORTED_SPEECH_MODELS = [
+        "default",
+        "numbers_and_commands",
+        "phone_call"
+    ]
+
+    @classmethod
+    def name(cls) -> Text:
+        """Name of your custom channel."""
+        return "twilio_voice"
+
+    @classmethod
+    def from_credentials(cls, credentials: Optional[Dict[Text, Any]]) -> InputChannel:
+        """Load custom configurations."""
+        credentials = credentials or {}
+
+        return cls(
+            credentials.get("initial_prompt", "hello"),
+            credentials.get("reprompt_fallback_phrase", "I'm sorry I didn't get that could you rephrase."),
+            credentials.get("assistant_voice", "woman"),
+            credentials.get("speech_timeout", "5"),
+            credentials.get("speech_model", "default"),
+            credentials.get("enhanced", "false")
+        )
+
+    def __init__(
+        self,
+        initial_prompt: Optional[Text],
+        reprompt_fallback_phrase: Optional[Text],
+        assistant_voice: Optional[Text],
+        speech_timeout: Optional[Text],
+        speech_model: Optional[Text],
+        enhanced: Optional[Text]
+    ) -> None:
+        """Creates a connection to Twilio voice.
+
+        Args:
+            assistant_voice: name of the assistant voice to use.
+        """
+        self.initial_prompt = initial_prompt
+        self.reprompt_fallback_phrase = reprompt_fallback_phrase
+        self.assistant_voice = assistant_voice
+        self.speech_timeout = speech_timeout
+        self.speech_model = speech_model
+        self.enhanced = enhanced
+
+        if assistant_voice not in self.SUPPORTED_VOICES:
+            self.raise_invalid_voice_exception()
+
+        try:
+            int(self.speech_timeout)
+        except ValueError:
+            if self.speech_timeout.lower() != "auto":
+                self.raise_invalid_speech_timeout_exception()
+
+        if speech_model not in self.SUPPORTED_SPEECH_MODELS:
+            self.raise_invalid_speech_model_exception()
+
+        if enhanced.lower() not in ["true", "false"]:
+            self.raise_invalid_enhanced_option_exception()
+
+        if (enhanced.lower() == "true") & (speech_model.lower() != "phone_call"):
+            self.raise_invalid_enhanced_speech_model_exception()
+
+        if (speech_model.lower() != "numbers_and_commands") & (speech_timeout.lower() == "auto"):
+            self.raise_invalid_speech_model_timeout_exception()
+
+    def raise_invalid_speech_model_timeout_exception(self) -> None:
```

Comment:
they can in my opinion all be protected as no code outside of the class would use it
```suggestion
    def _raise_invalid_speech_model_timeout_exception(self) -> None:
```